criu/files-reg.c

```diff
@@ -1629,6 +1739,47 @@ static int store_validation_data_build_id(RegFileEntry *rfe, int lfd,
 	return 1;
 }
 
+/*
+ * Finds and stores the CRC32C checksum of a file so that it can be validated
+ * while restoring.
+ * Returns 1 if the checksum of the file could be stored, -1 if there was an error
+ * or 0 if the checksum could not be obtained.
+ */
+static int store_validation_data_checksum(RegFileEntry *rfe, int lfd,
+						const struct fd_parms *p)
+{
+	u32 checksum;
+	int fd;
+
+	if (!p->stat.st_size)
+		return 0;
+
+	fd = open_proc(PROC_SELF, "fd/%d", lfd);
+	if (fd < 0) {
+		pr_err("Checksum (For validation) could not be obtained for file %s because can't open the file\n",
+				rfe->name);
+		return -1;
+	}
+
+	if (!calculate_checksum(fd, &(p->stat), &checksum)) {
+		close(fd);
+		pr_info("Could not obtain checksum for file %s\n", rfe->name);
+		return 0;
```

Comment:
Why do we not report an error in this case?

Reply:
I thought it would be better to inform the user that the checksum couldn't be obtained and only file size validation is being used for a particular file and proceed, instead of throwing an error and stopping. Should I change this?

Reply:
Yes, I think you should. If a user asks to check a checksum, they want to be sure that we will do this.